lib/PropertyEncoder.php

```diff
@@ -29,6 +29,24 @@ public function __construct(NamespaceRegistry $namespaceRegistry)
         $this->namespaceRegistry = $namespaceRegistry;
     }
 
+    public function encode($encoding, $name, $locale)
+    {
+        switch ($encoding) {
+            case 'system_localized':
+                return $this->localizedSystemName($name, $locale);
+            case 'system':
+                return $this->systemName($name);
+            case 'content_localized':
+                return $this->localizedContentName($name, $locale);
+            case 'content':
+                return $this->contentName($name);
+            default:
+                throw new \InvalidArgumentException(sprintf(
+                    'Invalid encoding "%s"', $encoding
```

Comment:
It would also be nice to make this extendable, would also need a `security` namespace here...